edge-modules/MetricsCollector/MetricWithCustomTags.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public sealed class MetricWithCustomTags : Metric, IEquatable<Metric>
```

Comment:
Why make this class instead of making metrics have a regular dictionary? Now you have to copy every metric just to add a tag.

Reply:
Lee, Philip, and I discussed in person. We did not want to make the original metric mutable since the tags are used to in the GetMetricKey() function. For the base metrics class, we settled on reverting the conversion to the tags dictionary in favor of keeping the original string type.